scripts/calculateRevenue/protocols/utils/networks.ts

```diff
@@ -0,0 +1,59 @@
+import { QueryResponse, TransactionField } from '@envio-dev/hypersync-client'
+import { getHyperSyncClient } from '../../../utils'
+import { NetworkId } from '../../../types'
+
+export async function fetchTotalGasUsed({
+  networkId,
+  users,
+  startBlock,
+  endBlock,
+}: {
+  networkId: NetworkId
+  users: string[]
+  startBlock?: number
+  endBlock?: number
+}): Promise<number> {
+  let fromBlock = startBlock ?? 0
+  let totalGasUsed = 0
+  let hasMoreBlocks = true
+
+  try {
+    const client = getHyperSyncClient(networkId)
+
+    const query = {
+      transactions: [{ from: users }],
+      fieldSelection: {
+        transaction: [TransactionField.GasUsed, TransactionField.GasPrice],
+      },
+      fromBlock,
+      ...(endBlock && { toBlock: endBlock }),
+    }
+
+    do {
```

Comment:
Can we just use `while` here? We never actually take advantage of the at-least-once execution semantics of `do while` since `hasMoreBlocks` will always be true initially.

Reply:
Changed in 1676e203764a05adeb799a89ebf8d5399dd40d3b.